File: Submissions/1302_Deepest_Leaves_Sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        int sum;
        while(!q.empty()) {
            sum = 0;
            int n = q.size();
            for(int i=0; i<n; ++i){
                TreeNode* current = q.front();
                sum += current->val;
                if(current->left) q.push(current->left);
                if(current->right) q.push(current->right);
                q.pop();
            }
        }
        return sum;
    }
};

File: Kotlin/1302_Deepest_Leaves_Sum.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun deepestLeavesSum(root: TreeNode?): Int {
        val queue: Queue<TreeNode> = LinkedList()
        var sum = 0
        queue.offer(root)
        
        while(queue.isNotEmpty()) {
            val n = queue.size
            sum = 0
            for(i in 0 until n) {
                val node = queue.poll()
                sum += node.`val`
                node.left?.let(queue::offer)
                node.right?.let(queue::offer)
            }
        }
    
        return sum
    }
}